test/e2e/gc_e2e_test.go

```diff
@@ -284,4 +285,104 @@ var _ = Describe("Garbage collection for dependent resources", func() {
 
 	})
 
+	Context("When installing a bundle with configmap and secret objects these objects should be deleted when the CSV is deleted", func() {
+		const (
+			sourceName    = "test-catalog"
+			imageName     = "quay.io/olmtest/single-bundle-index:objects"
+			packageName   = "busybox"
+			channelName   = "alpha"
+			subName       = "test-subscription"
+			csvName       = "busybox.v2.0.0"
+			secretName    = "mysecret"
+			configmapName = "special-config"
+		)
+
+		var (
+			source *v1alpha1.CatalogSource
+			err    error
+		)
+
+		BeforeEach(func() {
+			// create catalog source
+			source = &v1alpha1.CatalogSource{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       v1alpha1.CatalogSourceKind,
+					APIVersion: v1alpha1.CatalogSourceCRDAPIVersion,
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      sourceName,
+					Namespace: testNamespace,
+					Labels:    map[string]string{"olm.catalogSource": sourceName},
+				},
+				Spec: v1alpha1.CatalogSourceSpec{
+					SourceType: v1alpha1.SourceTypeGrpc,
+					Image:      imageName,
+				},
+			}
+
+			source, err = operatorClient.OperatorsV1alpha1().CatalogSources(source.GetNamespace()).Create(context.TODO(), source, metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred(), "could not create catalog source")
+
+			// wait for catalog pods to come up
+			selector := labels.SelectorFromSet(map[string]string{"olm.catalogSource": sourceName})
+			_, err = awaitPods(GinkgoT(), kubeClient, source.GetNamespace(), selector.String(), podCount(1))
+			Expect(err).ToNot(HaveOccurred(), "catalog source pod not found")
```

Comment:
Do we need to wait for the pod to come up? Won't the subscription eventually succeed?

Reply:
That's an interesting thought - we could create them in parallel. I'm a little concerned about that though because I believe `Eventually` has a default 60 second timeout - its possible on a slower cluster it could flake on CI. Let me see if I can set the timeout.

Reply:
Ok so I addressed this by relying on the default subscription checker polling which is 5 minutes. It did make the code cleaner this way, so nice catch.